zcash_client_backend/src/wallet.rs

```diff
@@ -119,9 +168,10 @@ pub struct WalletSaplingOutput<N> {
     is_change: bool,
     note_commitment_tree_position: Position,
     nf: N,
+    recipient_key_scope: S,
```

Comment:
The reason for that polymorphism (added in 8f03208439db1613739bb5bcf84e6791445088c5) was specifically to allow scanning with just `ivk`s (rather than `dfvk`s from which `ivk`s of known scope could be derived) while still enabling performance improvements. Put another way, the axis of polymorphism was whether we knew the scopes at all, not what protocol the scopes belonged to. So:
- I still think we should only use `zip32::Scope` for both Sapling and Orchard.
- Given that you want to report here the scope the output was received in, we could instead make this `recipient_key_scope: Option<zip32::Scope>` to allow for the case where we scanned with just `ivk`s with unknown scope (represented as `None`). But we can also leave in this genericism here, and just document the axis (i.e. that this will either be `zip32::Scope` or `()` depending on what kind of scanning was done).

Reply:
Using `Option<zip32::Scope>` complicates things because then we can't have the storage/retrieval methods typed to guarantee that a scope is always present. I don't mind documenting the scope of the variation, but TBH I think it's unnecessary, because the use sites all explicitly type it to `zip32::Scope` where a scope value is known to exist. Such documentation is also misleading, because it claims something that isn't enforced by the type, and therefor isn't an invariant - the caller can use whatever type they personally want for scope, so long as they "map" it to the desired type for the methods that require it to be so refined.